distribution/lib/Standard/Table/0.0.0-dev/src/IO/Auto_Detect.enso

```diff
@@ -1,36 +1,39 @@
-from Standard.Base import Any, Problem_Behavior, Nothing, Error, Panic, Meta, File, File_Format, Plain_Text, Bytes
-from Standard.Base.Error.Common import Unsupported_File_Type_Data, No_Such_Method_Error_Data, Illegal_Argument_Error_Data
+from Standard.Base import Any, Problem_Behavior, Nothing, Error, Panic, Meta, File, File_Format, Plain_Text_Format, Bytes
+from Standard.Base.Error.Common import Unsupported_File_Type, Unsupported_File_Type_Data, No_Such_Method_Error_Data, Illegal_Argument_Error_Data
 
-from Standard.Table.Delimited.Delimited_Format import Delimited
-from Standard.Table.Excel.Excel_Format import Excel
+import Standard.Table.Delimited.Delimited_Format
+import Standard.Table.Excel.Excel_Format
 
-types = [Plain_Text, Delimited ',', Excel, Bytes]
+## PRIVATE
+   Set of methods to get the configured File_Format for a file.
+format_providers = [Plain_Text_Format.for_file, Bytes.for_file, Delimited_Format.for_file, Excel_Format.for_file]
```

Comment:
This is a very minor thing but I'm thinking if we want to have `XYZ.for_file` here or just `XYZ` and make sure that `get_format` just delegates to `for_file` assuming a consistent interface?

An argument for _not_ having functions but modules instead on this list is that in the future we may introduce another function to the provider "interface" (oh how good it would be to have actual interfaces...) and then having modules here will naturally allow us to use this other method. With keeping callbacks, we don't have this kind of space for extensions. It also seems more natural to me with the plans for registering the providers - I'd rather register a provider instance than a callback function.